src/style-property-map-readonly.js

```diff
@@ -36,13 +36,12 @@
   };
 
   StylePropertyMapReadOnly.prototype.getProperties = function() {
-    var output = [];
-    for (var i = 0, l = this._styleObject.length; i < l; ++i) {
-      var property = this._styleObject[i];
-      // TODO: Construct the objects for each type of thing and add them to the
-      // output array.
+    var cssPropertiesFullText = this._styleObject.cssText.split('; ');
+    var cssProperties = [];
+    for (var i = 0; i < cssPropertiesFullText.length; i++) {
+      cssProperties[i] = cssPropertiesFullText[i].split(':')[0];
```

Comment:
Sorry I didn't notice this earlier - it's generally better to avoid parsing strings if you can, and you should only need to implement the getProperties in StylePropertyMapReadonly since StylePropertyMap inherits from it. If you move the logic from StylePropertyMap.prototype.getProperties to here, then everything should work as intended.

```javascript
>>> a = getComputedStyle($0)
CSSStyleDeclaration {0: "animation-delay", ... }
>>> a.length
264
>>> for (var i = 0; i < a.length; i++) { console.log(a[i]); }
animation-delay
animation-direction
animation-duration
...

>>> b = $0.style
CSSStyleDeclaration {alignContent: "", ...}
>>> b.length
0
>>> for (var i = 0; i < b.length; i++) { console.log(b[i]); }
undefined
```